src/parser.mly

```diff
@@ -0,0 +1,142 @@
+/* Ocamlyacc parser for SIFT */
+%{ 
+  open Ast 
+%}
+
+/* Arithmetic Operators */
+%token ADD SUBTRACT MULTIPLY DIVIDE MOD
+/* Relational Operators */
+%token EQ LEQ GT GEQ NEQ LT
+/* Logical Operators */
+%token AND OR NOT XOR
+/* Paranthesis and Brackets */
+%token LPAREN RPAREN LBRACE RBRACE LARROW RARROW LSQBRACE RSQBRACE
+/* Delimiters */
+%token COMMA SEMICOLON COLON DOT
+/* Assignment */
+%token ASSIGN 
+/* Features */
+%token PIPE
+/* Keywords */
+%token IF ELIF ELSE SWITCH CASE FOR WHILE CONTINUE BREAK DEFAULT LAMBDA DEFINE EXIT RETURN NULL TRUE FALSE
+%token IMPORT LIST FROM PURE 
+/* Primitive Data Types */
+%token INT FLOAT CHAR SYMBOL STRING BOOL
+/* Non-primitive Data Types */
+%token ARRAY LIST TUPLE SET DICT
+%token <int> INT_LITERAL
+%token <float> FLOAT_LITERAL
+%token <bool> BLIT
+%token <string> ID
+%token EOF
+
+
+%left SEMICOLON
+%nonassoc RSQBRACE RBRACE RARROW RPAREN
+%nonassoc LSQBRACE LBRACE LARROW LPAREN
+%nonassoc COLON
+%nonassoc COMMA
+%nonassoc NOELSE
```

Comment:
What is `NOELSE`?

Reply:
To handle dangling else problem. Like C does